Restructuring/projects handout (#414)
neuefische/web-exercises: sessions/recap-project-5/art-gallery-app/README.md

# Recap Project 5: Art Gallery

In this project, your team will create an interactive art gallery app.

This is a group project, so please create **only one project per group** following the [guide in the template section](./README.md#template).

## Overview

This project involves creating an art gallery app with four main pages: **Spotlight**, **Gallery**, **Favorite**, and **Detail**. Users will have the ability to like or unlike art pieces, and all liked pieces will appear on the **Favorite** page.

Each item from the API includes a URL to the art piece image. Use the `next/image` component to display these images. Remember to add `example-apis.vercel.app` to the allowed domains in the Next.js configuration file. You can refer to the **Next.js Basics and Routing** session handout for guidance. (Don’t forget to restart the development server after making any configuration changes.)

In addition to displaying the art pieces, your gallery app should handle dynamic data that updates based on user interactions. You’ll need to establish a state to manage information linked to the art pieces from the API. Refer to the **React Data Fetching** session handout for further guidance.

> The goal of this project is to write clean code with well-isolated components rather than to complete all features. Follow the outlined tasks within each user story for structure.

Reviewer: ```suggestion
The goal of this project is to write clean code with well-isolated components rather than to complete all features. Follow the outlined tasks within each user story to get on track.
```

Plan your work effectively, and coordinate your next steps as a team.

## API

We will fetch all necessary information about the art pieces from an API specifically designed for this course. [Check out the API](https://example-apis.vercel.app/api/art).

```bash
https://example-apis.vercel.app/api/art
```

## Template

1. Open your terminal and navigate to your projects folder.
2. Execute the following command to create a new project based on a template:

   ```bash
   npx ghcd@latest neuefische/web-exercises/tree/main/templates/next art-gallery-app -i
   ```

3. Create a new empty repository on GitHub and add all group members as collaborators.
4. Link your local repository to the GitHub repository and push the code.
5. Each group member should clone the repository to their local machine.
6. Follow the instructions in the `README.md` file.

> > 💡 **Tip**: Work on feature branches to avoid merge conflicts!

Reviewer: ```suggestion
> 💡 **Reminder**: Work on feature branches to avoid merge conflicts!
```
